RFC.md

```diff
@@ -1,19 +1,171 @@
 # RFC: Amplify JS TypeScript Improvements
 
 # General TypeScript Improvements
+
 - Version upgrade
 - Strict mode
 
 # Utility Changes
-- Hub typing
-- Configuration
+
+Amplify is proposing the following changes for our core utilities.
+
+## TypeScript support for Amplify Hub channels
+
+We are improving **_DX_** by adding strict type support to Hub channels, events, and payloads. An example of the developer experience when listening for `auth` events is highlighted below.
+
+**Amplify v5 & v6 (`aws-amplify@^5`)**
+
+```Typescript
+const channel = "auth"
+const signInFailureEvent = "signInFailure"
+const signInFailureData = new AuthError("sign-in failed")
+
+// It doesn't show a list of all supported channels, and it is not able to infer the right data given an event.
+Hub.dispatch("auth", {
+  event: "signInFailure",
+  data: new AuthError("Sign-in failed"),
+})
```

Comment:
it feels odd to have `Hub.dispatch("auth"...)` why do you need that?

Reply:
we dispatch `auth` events